swap/swap.go

```diff
@@ -344,6 +368,42 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 	return err
 }
 
+func (s *Swap) handleConfirmChequeMsg(ctx context.Context, p *Peer, msg *ConfirmChequeMsg) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	cheque := msg.Cheque
+
+	if p.getPendingCheque() == nil {
+		p.logger.Warn("ignoring confirm msg, no pending cheque")
+	}
+
+	if !cheque.Equal(p.getPendingCheque()) {
+		p.logger.Warn("ignoring confirm msg, wrong cheque", "got", cheque, "expected", p.getPendingCheque())
+		return
+	}
+
+	if err := p.setLastSentCheque(cheque); err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	if err := p.setPendingCheque(nil); err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	if err := p.updateBalance(int64(cheque.Honey)); err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	// since more swap traffic might have occurred since this cheque was already sent, we redo the payment threshold check
+	if err := s.checkPaymentThreshold(p); err != nil {
```

Comment:
If `sendCheque` returns an error here, it is handled differently than when `sendCheque` returns an error while being called from the `Add` function. I don't think this is desirable.

Reply:
I agree. I intentionally didn't change the error handling in `Add` in this PR to keep it simple. `Add` shouldn't return an error either if cheque sending fails but I would leave that (and the rest of the errors issues in `Add`) to a separate PR and only deal with pending cheques in this one.